pkg/compactor/compactor.go

```diff
@@ -59,7 +59,11 @@ func (cp *Compactor) Compact(ro *brtypes.RestoreOptions, needDefragmentation boo
 	}
 
 	// Set a temporary etcd data directory for embedded etcd
-	cmpctDir, err := ioutil.TempDir("/tmp", "compactor-")
+	prefix := cmpctOptions.Config.RestoreDataDir
+	if prefix == "" {
+		prefix = "/tmp"
+	}
+	cmpctDir, err := ioutil.TempDir(prefix, "compactor-")
```

Comment:
We should check whether the `prefix` path exists. `ioutils.TempDir` could potentially throw up an error if it's not a valid dir

Reply:
That's fine. Even if the path doesn't exist, compaction still should throw error